libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/ranker/DataLayoutStrategyScorer.java

```diff
@@ -0,0 +1,15 @@
+package com.linkedin.openhouse.datalayout.ranker;
+
+import com.linkedin.openhouse.datalayout.strategy.DataLayoutStrategy;
+import java.util.List;
+
+/** Interface for scoring data layout strategies. */
+public interface DataLayoutStrategyScorer {
+  /**
+   * Compute scores for the data layout strategies based on the input data.
+   *
+   * @param dataLayoutStrategies the data layout strategies to score
+   * @return the data layout strategies w/ computed scores
+   */
+  List<DataLayoutStrategy> scoreDataLayoutStrategies(List<DataLayoutStrategy> dataLayoutStrategies);
```

Comment:
We can reuse DataLayoutStrategy for now. DataLayoutStrategy should probably only store raw traits: file counts, compute cost, file skip counts (in the future). The method would return an entity like
```
class ScoredDataLayoutStrategy {
   DataLayoutStrategy strategy;
   double score;
   double cost;
}
```
We can also put weights and normalized values in this class but those are more volatile whereas I think score and cost are pretty much permanent values for a scored strategy.

Reply:
make sense.